### ISSUE TITLE
prefix argument in loggers

### ISSUE THREAD

**Issue by rohitgr7** [PyTorchLightning/pytorch-lightning#4537]
## 🚀 Feature
A simple `prefix` argument in loggers, similar to what we have in [ModelCheckpoint](https://github.com/PyTorchLightning/pytorch-lightning/blob/51cc7a89ee9f1c0ad35c147ce37a5825ec6d77e5/pytorch_lightning/callbacks/model_checkpoint.py#L145) that can prepend this value in metric names.

### Motivation
One use-case where I need this is while doing Kfold training. For now, I have to do it manually by updating the metric_name in `self.log`, but this can be done by the loggers using this `prefix` parameter. There might be other use-cases too.

### Pitch
if `prefix='fold1'`
and we do self.log('val_loss', ...)
logged metric will be `fold1_val_loss`.

Let me know if this is a good feature to add :)

cc @PyTorchLightning/core-contributors

**Comment by SkafteNicki** [comment]
I like this feature 👍 
Additionally, we should consider also adding a similar class variable as `CHECKPOINT_JOIN_CHAR`.
